src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/mirrorpoint/delete/actions/ValidateRequestAction.java

```diff
@@ -0,0 +1,112 @@
+import org.openkilda.model.PathId;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.persistence.repositories.FlowMirrorPathRepository;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.wfm.CommandContext;
+import org.openkilda.wfm.share.history.model.FlowEventData;
+import org.openkilda.wfm.share.logger.FlowOperationsDashboardLogger;
+import org.openkilda.wfm.topology.flowhs.exception.FlowProcessingException;
+import org.openkilda.wfm.topology.flowhs.fsm.common.actions.NbTrackableAction;
+import org.openkilda.wfm.topology.flowhs.fsm.mirrorpoint.delete.FlowMirrorPointDeleteContext;
+import org.openkilda.wfm.topology.flowhs.fsm.mirrorpoint.delete.FlowMirrorPointDeleteFsm;
+import org.openkilda.wfm.topology.flowhs.fsm.mirrorpoint.delete.FlowMirrorPointDeleteFsm.Event;
+import org.openkilda.wfm.topology.flowhs.fsm.mirrorpoint.delete.FlowMirrorPointDeleteFsm.State;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Optional;
+
+@Slf4j
+public class ValidateRequestAction
+        extends NbTrackableAction<FlowMirrorPointDeleteFsm, State, Event, FlowMirrorPointDeleteContext> {
+    private final FlowOperationsDashboardLogger dashboardLogger;
+    private final FlowMirrorPathRepository flowMirrorPathRepository;
+
+    public ValidateRequestAction(PersistenceManager persistenceManager,
+                                 FlowOperationsDashboardLogger dashboardLogger) {
+        super(persistenceManager);
+        RepositoryFactory repositoryFactory = persistenceManager.getRepositoryFactory();
+        flowMirrorPathRepository = repositoryFactory.createFlowMirrorPathRepository();
+        this.dashboardLogger = dashboardLogger;
+    }
+
+    @Override
+    protected Optional<Message> performWithResponse(State from, State to, Event event,
+                                                    FlowMirrorPointDeleteContext context,
+                                                    FlowMirrorPointDeleteFsm stateMachine) {
+        String flowId = stateMachine.getFlowId();
+        PathId mirrorPathId = new PathId(context.getFlowMirrorPointId());
+        stateMachine.setMirrorPathId(mirrorPathId);
+
+        dashboardLogger.onFlowMirrorPointDelete(flowId, context.getFlowMirrorPointId());
+
+        FlowMirrorPointResponse response = transactionManager.doInTransaction(() -> {
+            Flow foundFlow = getFlow(flowId);
+            if (foundFlow.getStatus() == FlowStatus.IN_PROGRESS) {
+                throw new FlowProcessingException(ErrorType.REQUEST_INVALID,
+                        format("Flow %s is in progress now", flowId));
+            }
+
+            FlowMirrorPath flowMirrorPath = flowMirrorPathRepository.findById(mirrorPathId)
+                    .orElseThrow(() -> new FlowProcessingException(ErrorType.NOT_FOUND,
+                            format("Flow mirror point %s not found", mirrorPathId)));
+            if (flowMirrorPath.getStatus() == FlowPathStatus.IN_PROGRESS) {
```

Comment:
What will happen if target flow will be changed while we are performing some action on mirror point?

Reply:
Now the flow is set to IN_PROGRESS status